src/lib/kernel.c

```diff
@@ -150,6 +150,36 @@ static const char *const s_koops_suspicious_strings[] = {
     NULL
 };
 
+static const char *const s_koops_suspicious_strings_blacklist[] = {
+    /* "BUG:" and "DEBUG:" overlaps, we don't want to recognize DEBUG messages as BUG */
+    "DEBUG:",
+
+    /* Termination */
+    NULL
+};
+
+static bool suspicious_line(const char *line)
+{
+    bool in_suspicious_list = false;
+    for (const char *const *str = s_koops_suspicious_strings; *str; ++str)
+        if (strstr(line, *str))
+        {
+            in_suspicious_list = true;
+            break;
+        }
+
```

Comment:
Just a side note. I believe that fast returns would make the code more readable:
```c
const char *const *str = s_koops_suspicious_strings;
for ( ; *str; ++str)
    if (strstr(line, *str))
        break;

if (!*str)
    return false;

str = s_koops_suspicious_strings_blacklist
for ( ; *str; ++str)
    if (strstr(line, *str))
       break;

return !!*str;
```